src/gg20/crypto_tools/paillier/zk/composite_dlog.rs

```diff
@@ -109,8 +120,14 @@ impl CompositeDLogStmt {
                 continue;
             }
 
-            // Sample s from {0,..,S-1} which is in Z*_N~ with high probability
+            // Sample s from {0,..,S-1} which is in Z*_N with high probability
             let s = Randomness::generate_with_rng(rng, &S);
+            debug_assert!(member_of_mul_group(&s.0, n));
+            if !member_of_mul_group(&s.0, n) {
```

Comment:
I realized my comment was incorrect. Since we're computing `s^(-1) mod phi(N)`, you want `s` to be invertible `mod phi(N)` and not `mod N`. I'll update it.